torch/_inductor/utils.py

```diff
@@ -2432,11 +2432,27 @@ def register_op_dtype_propagation_rules(
     )
 
 
+def get_current_backend() -> str:
+    from torch._inductor.virtualized import V
+
+    # cpp backend doesnt set current device - TODO: fix
+    if V.graph.current_device is None:
+        return "cpp"
```

Comment:
this seems a little fragile. Can we set the device instead ? i'm worried that there will be instances where we don't set the device in cuda and end up with cpp.

Reply:
yea, if you dont mind giving it a try

Reply:
Should be fine now with 92c5c6bddc73efe005424451f627bf6b46f44379